drivers/amplifiers/ada4250/ada4250.c

```diff
@@ -151,10 +151,31 @@ int32_t ada4250_soft_reset(struct ada4250_dev *dev)
 			return ret;
 
 		if(!(data & ADA4250_RESET(ADA4250_RESET_ENABLE)))
-			return SUCCESS;
+			break;
 	}
 
-	return FAILURE;
```

Comment:
I think we still need to return `FAILURE` in case of timeout. I suggest making a function that does the descriptor update by itself and calling it in the `if` clause above and keep the returns.

Reply:
Will follow this approach.